src/components/tl/mlx5/tl_mlx5.c

```diff
@@ -92,6 +92,10 @@ static ucc_config_field_t ucc_tl_mlx5_lib_config_table[] = {
      ucc_offsetof(ucc_tl_mlx5_lib_config_t, mcast_conf.max_eager),
      UCC_CONFIG_TYPE_MEMUNITS},
 
+    {"MCAST_CUDA_MEM_ENABLE", "0", "Enable GPU CUDA memory support for Mcast. GPUDirect RDMA must be enabled",
```

Comment:
Currently, if the cuda support for MCAST is asked from the team, then we also pre-post the buffers from GPU memort.  This is because I believe cudaMempcy for D2D is faster than H2D. We need this copy when we get a UD MCAST packet and we want to place it into app buffer.

Reply:
sure, but even if you have host buffers preposted you can still use tl mlx5 mcast for cuda buffers

Reply:
yes, we can use it, but since the performance will not be optimal, I rather rely on tl ucp to give us better performance. At very large scale though, using mcast even with staging into host buffer might be beneficial. For next PRs, I will make the changes so that it works even if we do not have GPUDirectRDMA.  Another idea is to create two mcast comm for each team (one for cuda, one for host). Therefore, we can guarantee best performance for both memory type (this requires GPUDirectRDMA).